app/lib/actions/pulp/repository/refresh.rb

```diff
@@ -4,80 +4,16 @@ module Repository
       class Refresh < Pulp::Abstract
         input_format do
           param :capsule_id
+          param :pulp_id
         end
 
         def plan(repository, input = {})
-          input[:capsule_id] ||= SmartProxy.default_capsule!.id
-          repository_details = pulp_extensions(input[:capsule_id]).repository.retrieve_with_details(repository.pulp_id)
-          update_or_associate_importer(input[:capsule_id], repository, repository_details)
-          update_or_associate_distributors(input[:capsule_id], repository, repository_details)
-          remove_unnecessary_distributors(input[:capsule_id], repository, repository_details)
+          plan_self(:capsule_id => input[:capsule_id], :pulp_id => repository.pulp_id)
         end
 
-        def update_or_associate_importer(capsule_id, repository, repository_details)
-          existing_importers = repository_details["importers"]
-          importer = repository.generate_importer(::SmartProxy.find_by(:id => capsule_id))
```

Comment:
This looks like the last instances of Repository#generate_importer and Repository#generate_distributors in the glue/pulp/repo.rb file.  Mind removing those methods and their tests?